engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/ModCheckBase.java

```diff
@@ -79,11 +82,22 @@ public boolean isValid(final CharSequence value, final ConstraintValidatorContex
 
 	public abstract boolean isCheckDigitValid(List<Integer> digits, char checkDigit);
 
-	protected void initialize(int startIndex, int endIndex, int checkDigitIndex, boolean ignoreNonDigitCharacters) {
+	protected void initialize(int startIndex, int endIndex, int checkDigitIndex, boolean ignoreNonValidCharacters) {
 		this.startIndex = startIndex;
 		this.endIndex = endIndex;
 		this.checkDigitIndex = checkDigitIndex;
-		this.ignoreNonDigitCharacters = ignoreNonDigitCharacters;
+		this.ignoreNonValidCharacters = ignoreNonValidCharacters;
+		this.alphanumeric = false;
+
+		this.validateOptions();
+	}
```

Comment:
maybe just call 
```java
initialize(startIndex, endIndex, checkDigitIndex, ignoreNonValidCharacters, false)
```

and also if you'll end up still needing to modify the `ModCheckBase` consider using this overloaded version of initialize in places where `alphanumeric` is not needed so that less classes are modified.